Bug: Fixture in `bgp/test_bgp_router_id.py` assumes VoQ is multi-asic causing failures

### Is it platform specific

generic

### Importance or Severity

Critical

### Description of the bug

Fixture assumes VoQ is multi-asic causing failures on single-asic:
```
duthosts = [<MultiAsicSonicHost str5-nh5010-ut2-5>], enum_frontend_dut_hostname = 'str5-nh5010-ut2-5', enum_frontend_asic_index = None, loopback_ip = '10.1.0.1'

    @pytest.fixture()
    def default_bgp_router_id(duthosts, enum_frontend_dut_hostname, enum_frontend_asic_index, loopback_ip):
        duthost = duthosts[enum_frontend_dut_hostname]
        cfg_facts = get_asic_config_facts(duthost, enum_frontend_asic_index)
        dev_meta = cfg_facts.get('DEVICE_METADATA', {}).get('localhost', {})

        if dev_meta.get('switch_type') in ['voq', 'chassis-packet']:
            host_vars = get_host_visible_vars(duthost.host.options['inventory'], duthost.hostname)
            loopback4096_ips = host_vars.get('loopback4096_ip', [])
>           if len(loopback4096_ips) > enum_frontend_asic_index:
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           TypeError: '>' not supported between instances of 'int' and 'NoneType'
```

Introduced in https://github.com/sonic-net/sonic-mgmt/pull/25683

### Steps to Reproduce

Run test on single-asic VoQ

### Actual Behavior and Expected Behavior

Check if VoQ is multi-asic before assuming `enum_frontend_asic_index` is not None

### Relevant log output

```shell
N/A
```

### Output of `show version`

```shell
N/A
```

### Attach files (if any)

N/A

--- a/tests/bgp/test_bgp_router_id.py
+++ b/tests/bgp/test_bgp_router_id.py
@@ -137,7 +137,7 @@ def default_bgp_router_id(duthosts, enum_frontend_dut_hostname, enum_frontend_as
     cfg_facts = get_asic_config_facts(duthost, enum_frontend_asic_index)
     dev_meta = cfg_facts.get('DEVICE_METADATA', {}).get('localhost', {})
 
-    if dev_meta.get('switch_type') in ['voq', 'chassis-packet']:
+    if dev_meta.get('switch_type') in ['voq', 'chassis-packet'] and duthost.is_multi_asic:
         host_vars = get_host_visible_vars(duthost.host.options['inventory'], duthost.hostname)
         loopback4096_ips = host_vars.get('loopback4096_ip', [])
         if len(loopback4096_ips) > enum_frontend_asic_index:

